docs/AllBridgeFacet.md

```diff
@@ -0,0 +1,82 @@
+
+One feature that makes AllBridge different from other bridges is that it uses pools on either side of the bridge. This allows them to no having to have canonical representation on the destination chain. Thus for EVM based chain this means that the pool needs to be approved to spend the bridged tokens.
+
+Underneath, AllBridge can use different message protocols to transfer tokens. These are represented in an enum like
+
+```
+/// @title AllBridge Messenger Protocol Enum
+enum MessengerProtocol {
+    None,
+    Allbridge,
+    Wormhole,
+    LayerZero
+}
+```
+
+```mermaid
+graph LR;
+    D{LiFiDiamond}-- DELEGATECALL -->C(AllBridgeFacet);
+    C(AllBridgeFacet) -- CALL --> C(AllBridge Core)
+```
+
+## Public Methods
+
+- `function startBridgeTokensViaBridgeFacet(BridgeData calldata _bridgeData, AllBridgeData calldata _allBridgeData)`
+  - Simply bridges tokens using bridgeFacet
+- `function swapAndStartBridgeTokensViaBridgeFacet(BridgeData memory _bridgeData, LibSwap.SwapData[] calldata _swapData, AllBridgeData calldata _allBridgeData)`
+  - Performs swap(s) before bridging tokens using bridgeFacet
+
+## Bridge Specific Parameters
+
+Some of the methods listed above take a variable labeled `_allBridgeData`.
+
+This data is specific to allBridgefacet and is represented as the following struct type:
+
+```solidity
+/// @notice The struct for the AllBridge data.
+/// @param fees The amount of token to pay the messenger and the bridge
+/// @param recipient The address of the token receiver after bridging.
+/// @param destinationChainId The destination chain id.
+/// @param receiveToken The token to receive on the destination chain.
+/// @param nonce A random nonce to associate with the tx.
+/// @param messenger The messenger protocol enum
+struct AllBridgeData {
+  uint256 fees;
+  bytes32 recipient;
+  uint8 destinationChainId;
+  bytes32 receiveToken;
+  uint256 nonce;
+  MessengerProtocol messenger;
+}
+
+```
+
+The fees field is the sum of two fees charged by AllBridge, namely
+
+- MessengerFee: Fee charged by the underlying message layer. Parts of the messenger fee should cover relay fees as well.
+- AllBridgeFee. Fee charged by AllBridge itself.
+
+## Swap Data
+
+Some methods accept a `SwapData _swapData` parameter.
+
+Swapping is performed by a swap specific library that expects an array of calldata to can be run on variaous DEXs (i.e. Uniswap) to make one or multiple swaps before performing another action.
+
+The swap library can be found [here](../src/Libraries/LibSwap.sol).
+
+## LiFi Data
+
+Most of the methods accept a `ILiFiBridgeData _bridgeData` parameter.
+
+In the AllBridge contract call the fields `minAmount` and `sendingAssetId` are used for the transfer amount and the asset to be sent. Since the AllBridge bridge does not support native token briding (it's mainly a stablecoin bridge) the methods will failed if native assets is tried bridged.
```

Comment:
typos: 
"briding" => "bridging"
"will failed if native assets is tried bridged" => "will fail if native assets are tried to be bridged"

Reply:
👍